clippy_lints/src/doc.rs

```diff
@@ -747,17 +747,20 @@ fn check_word(cx: &LateContext<'_>, word: &str, span: Span) {
         }
     }
 
-    // We assume that mixed-case words are not meant to be put inside bacticks. (Issue #2343)
+    // We assume that mixed-case words are not meant to be put inside backticks. (Issue #2343)
     if has_underscore(word) && has_hyphen(word) {
         return;
     }
 
     if has_underscore(word) || word.contains("::") || is_camel_case(word) {
-        span_lint(
+        span_lint_and_sugg(
             cx,
             DOC_MARKDOWN,
             span,
-            &format!("you should put `{}` between ticks in the documentation", word),
+            "item in documentation missing backticks",
+            "try",
+            format!("`{}`", snippet(cx, span, "..")),
```

Comment:
Here it would be better to use `snippet_with_applicability` to ensure that the suggestion is only `MachineApplicable` if the snipped was extracted correctly :upside_down_face: 

[Here is an example usage of `snippet_with_applicability`](https://github.com/rust-lang/rust-clippy/blob/1f955158ddc2035a054fdd4d7cc3bd2297b5a7ba/clippy_lints/src/len_zero.rs#L420-L433)